test/jdk/com/sun/jdi/TestScaffold.java

```diff
@@ -710,6 +710,25 @@ public synchronized void waitForVMDisconnect() {
             } catch (InterruptedException e) {
             }
         }
+
+        // Make sure debuggee exits with no errors. Otherwise failures might go unnoticed.
+        Process p = vm.process();
+        try {
+            p.waitFor(5, java.util.concurrent.TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        if (p.isAlive()) {
```

Comment:
It think it is better to just use watiFor(). If debugee hangs it would be better to times out and give timeout handler a chance to dump all stack traces.

Reply:
Ok. I hadn't thought about the timeout handler.